File: leetcode/JumpGame.java
class Solution {
    public boolean canJump(int[] nums) {
        
        if (nums.length < 1 ){
            return false;
        }
        
        if (nums.length == 1){
            return true;
        }
        
        int lastPos = nums.length - 1;
        
        for (int i = nums.length -2; i >= 0; i--){
            if (nums[i] + i >= lastPos){
                lastPos = i;
            }
        }
        
        if (lastPos == 0){
            return true;
        } else {
            return false;
        }
        
    }
}

File: leetcode/JumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        
        if (len(nums) == 1):
            return True
        
        lastPos = len(nums) - 1
        
        for i in range(len(nums)-2, -1, -1):
            
            if (i + nums[i] >= lastPos):
                lastPos = i
                
        if (lastPos == 0):
            return True
        
        else:
            return False
            
        